### ISSUE TITLE
HTTP/3: SocketsHttpHandler.PlaintextStreamFilter

### ISSUE THREAD

**Issue by wfurt** [dotnet/runtime#58089]
This is similar to #48617 but the motivation is different. Main driver for this is ability to debug HTTP3 and anything inside like gRPC or custom application data.
While creating custom transport may be quite challenging with QUIC transport, ability to dump decrypted data should be relatively simple and completely within HttpClient scope.

**Comment by geoffkizer** [comment]
Agreed, we should just do this. We should wrap every HTTP3 request stream with a call to the PlaintextStreamFilter if specified.

**Comment by ManickaP** [comment]
As agreed in our discussion and noted here: https://github.com/dotnet/runtime/issues/64449. We decided to postpone `PlaintextStreamFilter` to the later release since for this to work we'd need either `QuicStream` abstraction or these 2 `Stream` issues: https://github.com/dotnet/runtime/issues/43290 and https://github.com/dotnet/runtime/issues/25344.
Punting to Future column for now.
